### PR TITLE
Update v2 bundle spec

### PR DESCRIPTION
**Pull request by jenaepennie**
Shows lab author how to specify multiple open-console buttons within in lab control panel

### CODE REVIEW

**Review by robsideshow — COMMENTED** [review]
I think you should add a sentence about this change to the [Changelog](https://github.com/CloudVLab/qwiklabs-content-bundle-spec/blob/f2966cc9c8d757951a03aa161b9a4869507ae639/lab-bundle-spec.md#changelog) section where it talks about `student_visible_outputs`.  This is an important feature!

I would also edit the [gcp references](https://github.com/CloudVLab/qwiklabs-content-bundle-spec/blob/f2966cc9c8d757951a03aa161b9a4869507ae639/lab-bundle-spec.md#valid-resource-references) and [aws references](https://github.com/CloudVLab/qwiklabs-content-bundle-spec/blob/f2966cc9c8d757951a03aa161b9a4869507ae639/lab-bundle-spec.md#valid-custom-property-references) sections, pointing out which references are buttons and which are copyable text.  In the aws section, it might be nice to rearrange the order, so the buttons are in a group of 3 and the rest are in a second group.

File: lab-bundle-spec.md [hunk 1]
```diff
@@ -357,34 +364,58 @@ For a complete and up-to-date list of EC2 instance types, see [AWS official docu
 
 Specify which resource properties are given to the lab taker.
 
+For the lab taker to get access to a `gcp_project`, a `console_url` resource reference must be provided.
+For the lab taker to get access to an `aws_account`, one of the following resource references must be provided:
+- `console_url`
+- `sts_link`
+- `vnc_link`
+
+All `console_url`, `sts_link`, and `vnc_link` resource references are presented as an button to the lab taker.
```
Comment (robsideshow) [inline lab-bundle-spec.md:373]: (nit) typo: "an button" --> "a button"

**Review by robsideshow — COMMENTED** [review]
The example in https://github.com/CloudVLab/qwiklabs-content-bundle-spec/blob/master/examples/lab-minimal/v2/qwiklabs.yml  should probably have a `console_url` added to the student_visible_outputs`.

You might also mention explicitly that for the button outputs, the `label` provided by the author will be the exact text *on the button*.  Up to you if you think that helps.

**Comment by robsideshow** [comment]
@jgan @supersam654 @joncass

File: lab-bundle-spec.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ The primary changes made from v1 to v2 are:
 
 - Adding `environment` as a top level property with two children:
   - `resources` matches the usage of `environment_resources` in v1
-  - `student_visible_outputs` specifies which properties of those resources should be provided to the student (e.g. only display the username and password for the `primary_user` instead of every `gcp_user` created by the lab). See the [Student Visible Outputs section](#student-visible-outputs) for details.
+  - `student_visible_outputs` specifies which properties of those resources should be provided to the student (e.g. only display the username and password for the `primary_user` instead of every `gcp_user` created by the lab). All `console_url`, `sts_link`, and `vnc_link` resource references are displayed as a button while the rest of the resource references are displayed as copyable text. See the [Student Visible Outputs section](#student-visible-outputs) for details.
```
Comment (supersam654) [inline lab-bundle-spec.md:13]: Nit: "a button" -> "buttons"

File: lab-bundle-spec.md [hunk 3]
```diff
@@ -20,6 +20,8 @@ The primary changes made from v1 to v2 are:
 
 - Adding an `ssh_key_user` attribute on the `gcp_project` resource type.
 
+- Adding a `console_url` reference to the `gcp_project` resource type.
```
Comment (supersam654) [inline lab-bundle-spec.md:23]: I'd probably say something more broadly like "Requiring lab creators to specify exact control panel outputs with `student_visible_outputs`."

File: examples/lab-minimal/v2/qwiklabs.yml [hunk 4]
```diff
@@ -21,6 +21,10 @@ instruction:
 
 environment:
   student_visible_outputs:
+    - label:
+        locales:
+          en: "Open GCP Console"
+      reference: primary_project.console_url
```
Comment (supersam654) [inline examples/lab-minimal/v2/qwiklabs.yml:27]: While the definition of "minimal" is rather subjective, I'd recommend also including a gcp user username + password here as that's what a "typical" V1 converted lab would look like.

File: lab-bundle-spec.md [hunk 5]
```diff
@@ -335,17 +341,18 @@ Note that a lab will only launch on a given deployment if the deployment has an
...
-- [AWS_ACCOUNT].username
-- [AWS_ACCOUNT].password
-<!-- Legacy display option replacements -->
-- [AWS_ACCOUNT].sts_link
-- [AWS_ACCOUNT].access_key_id
-- [AWS_ACCOUNT].console_url
-- [AWS_ACCOUNT].rdp_credentials
-- [AWS_ACCOUNT].ssh_credentials
-- [AWS_ACCOUNT].vnc_link
+<!-- Legacy display option replacements after password resource reference-->
```
Comment (supersam654) [inline lab-bundle-spec.md:344]: This HTML comment doesn't make sense up here. I don't think it's super important to preserve so I'd recommend deleting it.

File: lab-bundle-spec.md [hunk 6]
```diff
@@ -357,34 +364,58 @@ For a complete and up-to-date list of EC2 instance types, see [AWS official docu
 
 Specify which resource properties are given to the lab taker.
 
+For the lab taker to get access to a `gcp_project`, a `console_url` resource reference must be provided.
```
Comment (supersam654) [inline lab-bundle-spec.md:367]: Nit: "provided" -> "specified"

Comment (supersam654) [reply lab-bundle-spec.md:367]: Same for next line.

File: lab-bundle-spec.md [hunk 7]
```diff
@@ -357,34 +364,58 @@ For a complete and up-to-date list of EC2 instance types, see [AWS official docu
...
+For the lab taker to get access to an `aws_account`, one of the following resource references must be provided:
+- `console_url`
+- `sts_link`
+- `vnc_link`
+
+All `console_url`, `sts_link`, and `vnc_link` resource references are presented as a button to the lab taker. The label provided with the resource reference will be displayed on the button.
```
Comment (supersam654) [inline lab-bundle-spec.md:373]: Nit: "lab taker" -> "student". We have enough nomenclature as-is and I'd rather not introduce more unless we really need to.

**Review by robsideshow — APPROVED** [review]
LGTM

File: examples/lab-minimal/v2/qwiklabs.yml (hunk 4 above)
Comment (robsideshow) [reply examples/lab-minimal/v2/qwiklabs.yml:27]: It does already, doesn't it?  Lines 32-39?

Comment (supersam654) [reply examples/lab-minimal/v2/qwiklabs.yml:27]: I could have sworn I looked down there but you're absolutely right.

File: lab-bundle-spec.md (hunk 7 above)
Comment (jenaepennie) [reply lab-bundle-spec.md]: updated

File: lab-bundle-spec.md (hunk 5 above)
Comment (jenaepennie) [reply lab-bundle-spec.md]: removed

File: lab-bundle-spec.md (hunk 3 above)
Comment (jenaepennie) [reply lab-bundle-spec.md]: updated

**Pull request merged**